### PR TITLE
5 DMU cards

### PR DESCRIPTION
**Pull request by squee1968**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/djinn_of_the_fountain.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+SVar:TrigCharm:DB$ Charm | Choices$ DBPump,DBExile,DBScry
+SVar:DBPump:DB$ Pump | Defined$ Self | NumAtt$ 1 | NumDef$ 1 | SpellDescription$ CARDNAME gets +1/+1 until end of turn.
+SVar:DBExile:DB$ ChangeZone | Defined$ Self | Origin$ Battlefield | Destination$ Exile | SubAbility$ DelTrig | RememberChanged$ True | SpellDescription$ Exile CARDNAME. Return it to the battlefield under its owner's control at the beginning of the next end step.
+SVar:DelTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | Execute$ TrigReturn | ConditionDefined$ Remembered | ConditionPresent$ Card | TriggerDescription$ Return CARDNAME to the battlefield.
+SVar:TrigReturn:DB$ ChangeZone | Defined$ Self | Origin$ Exile | Destination$ Battlefield
+SVar:DBScry:DB$ Scry | ScryNum$ 1 | SpellDescription$ Scry 1.
+DeckHints:Type$Instant|Sorcery
+SVar:BuffedBy:Spell.Instant,Spell.Sorcery
+Oracle:Flying\nWhenever you cast an instant or sorcery spell, choose one —\n• Djinn of the Fountain gets +1/+1 until end of turn.\n• Exile Djinn of the Fountain. Return it to thebattlefield under its owner's control at the beginning of the next end step.\n• Scry 1.
```
Comment (Agetian) [inline forge-gui/res/cardsfolder/upcoming/djinn_of_the_fountain.txt:15]: "the battlefield" (missing a space)

**Review by Agetian — DISMISSED** [review]

File: forge-gui/res/cardsfolder/upcoming/coral_colony.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Name:Coral Colony
+ManaCost:1 U
+Types:Creature Wall
+PT:1/4
+K:Defender
+A:AB$ Mill | Cost$ 1 U T | NumCards$ X | ValidTgts$ Player | TgtPrompt$ Choose a player | SpellDescription$ Target player mills X cards, where X is the number of cards you control with defender.
+SVar:X:Count$Valid Creature.withDefender+YouCtrl
+Oracle:Defender\n{1}{U}, {T}: Target player mills X cards, where X is the number of cards you control with defender. (To mill a card, a player puts the top card of their library into their graveyard.)
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/coral_colony.txt:8]: DeckHas:Ability$Mill
DeckHints:Keyword$Defender

File: forge-gui/res/cardsfolder/upcoming/djinn_of_the_fountain.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+PT:4/4
+K:Flying
+T:Mode$ SpellCast | TriggerZones$ Battlefield | ValidCard$ Instant,Sorcery | ValidActivatingPlayer$ You | NoResolvingCheck$ True | Execute$ TrigCharm | TriggerDescription$ Whenever you cast an instant or sorcery spell, ABILITY
+SVar:TrigCharm:DB$ Charm | Choices$ DBPump,DBExile,DBScry
+SVar:DBPump:DB$ Pump | Defined$ Self | NumAtt$ 1 | NumDef$ 1 | SpellDescription$ CARDNAME gets +1/+1 until end of turn.
+SVar:DBExile:DB$ ChangeZone | Defined$ Self | Origin$ Battlefield | Destination$ Exile | SubAbility$ DelTrig | RememberChanged$ True | SpellDescription$ Exile CARDNAME. Return it to the battlefield under its owner's control at the beginning of the next end step.
+SVar:DelTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | Execute$ TrigReturn | ConditionDefined$ Remembered | ConditionPresent$ Card | TriggerDescription$ Return CARDNAME to the battlefield.
+SVar:TrigReturn:DB$ ChangeZone | Defined$ Self | Origin$ Exile | Destination$ Battlefield
+SVar:DBScry:DB$ Scry | ScryNum$ 1 | SpellDescription$ Scry 1.
+DeckHints:Type$Instant|Sorcery
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/djinn_of_the_fountain.txt:13]: maybe upgrade to needs? This is pretty bad if it never triggers

File: forge-gui/res/cardsfolder/upcoming/fires_of_victory.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Name:Fires of Victory
+ManaCost:1 R
+Types:Instant
+K:Kicker:2 U
+A:SP$ DealDamage | ValidTgts$ Creature,Planeswalker | TgtPrompt$ Select target creature or planeswalker | NumDmg$ X | SubAbility$ DBDraw | SpellDescription$ If this spell was kicked, draw a card. CARDNAME deals damage to target creature or planeswalker equal to the number of cards in your hand.
+SVar:DBDraw:DB$ Draw | NumCards$ 1 | Condition$ Kicked | ConditionDescription$ If this spell was kicked
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/fires_of_victory.txt:6]: The Draw needs to be before the DealDamage

Comment (squee1968) [reply forge-gui/res/cardsfolder/upcoming/fires_of_victory.txt:6]: How and why? Do you mean just switch lines 5 and 6 with no changes? Or do I write it as A:SP$ Draw with the DealDamage as the SubAbility? If so, what happens if the spell isn't kicked?

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/fires_of_victory.txt]: if you do:
* Draw With Condition
* Deal Damage

then it should just work without being kicked.

@Agetian just might need to take a look later if AI works as it should

**Review by tool4ever — DISMISSED** [review]

File: forge-gui/res/cardsfolder/upcoming/djinn_of_the_fountain.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+Name:Djinn of the Fountain
+ManaCost:4 U U
+Types:Creature Djinn
+PT:4/4
+K:Flying
+T:Mode$ SpellCast | TriggerZones$ Battlefield | ValidCard$ Instant,Sorcery | ValidActivatingPlayer$ You | NoResolvingCheck$ True | Execute$ TrigCharm | TriggerDescription$ Whenever you cast an instant or sorcery spell, ABILITY
+SVar:TrigCharm:DB$ Charm | Choices$ DBPump,DBExile,DBScry
+SVar:DBPump:DB$ Pump | Defined$ Self | NumAtt$ 1 | NumDef$ 1 | SpellDescription$ CARDNAME gets +1/+1 until end of turn.
+SVar:DBExile:DB$ ChangeZone | Defined$ Self | Origin$ Battlefield | Destination$ Exile | SubAbility$ DelTrig | RememberChanged$ True | SpellDescription$ Exile CARDNAME. Return it to the battlefield under its owner's control at the beginning of the next end step.
+SVar:DelTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | Execute$ TrigReturn | ConditionDefined$ Remembered | ConditionPresent$ Card | TriggerDescription$ Return CARDNAME to the battlefield.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/djinn_of_the_fountain.txt:10]: I think make the TrigDesc closer to the Oracle... maybe:
`Return CARDNAME to the battlefield under its owner's control at the beginning of the next end step.`

**Review by Northmoc — DISMISSED** [review]
Agh, found one more thing to comment on. Great work @squee1968 !

**Review by tool4ever — APPROVED** [review]

**Pull request merged**